drivers/regulator/regulator_mspm0_vref.c

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright 2025 Linumiz GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_mspm0_vref
+
+#include <errno.h>
+#include <zephyr/drivers/regulator.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/dt-bindings/clock/mspm0_clock.h>
+#include <zephyr/dt-bindings/regulator/mspm0_vref.h>
+#include <zephyr/logging/log.h>
+
+/* TI Driverlib includes */
+#include <ti/driverlib/dl_vref.h>
+
+LOG_MODULE_REGISTER(vref, CONFIG_LOG_DEFAULT_LEVEL);
+
+#define VREF_1_4V	1400000
+#define VREF_2_5V	2500000
+
+struct regulator_mspm0_vref_data {
+	struct regulator_common_data common;
+	DL_VREF_Config vref_cfg;
+};
+
+struct regulator_mspm0_vref_config {
+	struct regulator_common_config common;
+	const struct pinctrl_dev_config *vref_pin;
+	DL_VREF_ClockConfig vref_clock_cfg;
+	VREF_Regs *regs;
+};
+
+static int regulator_mspm0_vref_enable(const struct device *dev)
+{
+	const struct regulator_mspm0_vref_config *config = dev->config;
+
+	DL_VREF_enableInternalRef(config->regs);
+
+	return 0;
+}
+
+static int regulator_mspm0_vref_disable(const struct device *dev)
+{
+	const struct regulator_mspm0_vref_config *config = dev->config;
+
+	DL_VREF_disableInternalRef(config->regs);
```

Comment:
Unless enabling and disabling HAL calls are atomic, you might want to protect it if CONFIG_REGULATOR_THREAD_SAFE_REFCNT is enabled